rapids_build_backend/impls.py

```diff
@@ -86,126 +91,9 @@ def _get_cuda_suffix(require_cuda=False):
         The CUDA suffix (e.g., "-cu11") or an empty string if CUDA could not be
         detected.
     """
-    if (major := _get_cuda_major(require_cuda)) is None:
+    if (version := _get_cuda_version(require_cuda)) is None:
         return ""
-    return f"-cu{major}"
-
-
-# Wheels with a CUDA suffix.
-_VERSIONED_RAPIDS_WHEELS = [
-    "rmm",
-    "pylibcugraphops",
-    "pylibcugraph",
-    "nx-cugraph",
-    "dask-cudf",
-    "cuspatial",
-    "cuproj",
-    "cuml",
-    "cugraph",
-    "cudf",
-    "ptxcompiler",
-    "cubinlinker",
-    "cugraph-dgl",
-    "cugraph-pyg",
-    "cugraph-equivariant",
-    "raft-dask",
-    "pylibwholegraph",
-    "pylibraft",
-    "cuxfilter",
-    "cucim",
-    "ucx-py",
-    "ucxx",
-    "pynvjitlink",
-    "distributed-ucxx",
-]
-
-# Wheels without a CUDA suffix.
-_UNVERSIONED_RAPIDS_WHEELS = [
-    "dask-cuda",
-    "rapids-dask-dependency",
-]
-
-# Wheels that don't release regular alpha versions
-_CUDA_11_ONLY_WHEELS = (
-    "ptxcompiler",
-    "cubinlinker",
-)
-
-
-def _add_cuda_suffix(req, cuda_suffix, cuda_major):
-    req = Requirement(req)
-    if req.name == "cupy" and cuda_major is not None:
-        req.name += f"-cuda{cuda_major}x"
-    elif req.name in _VERSIONED_RAPIDS_WHEELS:
-        req.name += cuda_suffix
-
-    return str(req)
-
-
-def _add_alpha_specifier(req):
-    req = Requirement(req)
-    if (
-        req.name in _VERSIONED_RAPIDS_WHEELS or req.name in _UNVERSIONED_RAPIDS_WHEELS
-    ) and req.name not in _CUDA_11_ONLY_WHEELS:
-        req.specifier &= SpecifierSet(">=0.0.0a0")
-    return str(req)
```

Comment:
We still want this logic as part of rbb, not in dependencies.yaml. I don't think we want to create separate dependency lists for nightly vs non-nightly dependencies in dependencies.yaml, but we also can't have the released versions of our packages contain the alpha spec or that could cause problems.